pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -137,64 +129,67 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
-		err := r.client.Create(context.TODO(), newCRB)
-		if err != nil {
-			groupPermissionConditions := instance.Status.Conditions
-
-			// make new condition
-			newCondition := managedv1alpha1.Condition{
-				LastTransitionTime: metav1.Now(),
-				ClusterRoleName:    clusterRoleName,
-				Status:             true,
-				State:              "Created",
-			}
-
-			// append new condition to conditions array
-			groupPermissionConditions = append(groupPermissionConditions, newCondition)
-
-			// update GroupPermission object with new conditions
-			instance.Status.Conditions = groupPermissionConditions
-
-			err = r.client.Status().Update(context.TODO(), instance)
+	allClusterRoleBindingsOnCluster, crClusterRoleBindingNameList := checkClusterRoleBindingNames(crClusterRoleBindingNameList, clusterRoleBindingList)
+	if !allClusterRoleBindingsOnCluster {
+		// loop through crClusterRoleBindingNameList
+		// make a newClusterRoleBinding for each one of them
+		// so newClusterRoleBinding should take in that name
+		for _, clusterRoleBindingName := range crClusterRoleBindingNameList {
+			// create a new clusterRoleBinding on cluster
+			newCRB := newClusterRoleBinding(clusterRoleBindingName)
+			err := r.client.Create(context.TODO(), newCRB)
 			if err != nil {
-				reqLogger.Error(err, "Failed to update condition.")
-				return reconcile.Result{}, err
-			}
-		} else {
-			groupPermissionConditions := instance.Status.Conditions
-
-			// make new condition
-			newCondition := managedv1alpha1.Condition{
-				LastTransitionTime: metav1.Now(),
-				Message:            "Unable to create ClusterRoleBinding: <error from platform>",
-				ClusterRoleName:    clusterRoleName,
-				Status:             true,
-				State:              "Failed",
-			}
-
-			// append new condition to conditions array
-			groupPermissionConditions = append(groupPermissionConditions, newCondition)
-
-			// update GroupPermission object with new conditions
-			instance.Status.Conditions = groupPermissionConditions
-
-			err = r.client.Status().Update(context.TODO(), instance)
-			if err != nil {
-				reqLogger.Error(err, "Failed to update condition.")
-				return reconcile.Result{}, err
+				groupPermissionConditions := instance.Status.Conditions
+
+				// make new condition
+				newCondition := managedv1alpha1.Condition{
+					LastTransitionTime: metav1.Now(),
+					ClusterRoleName:    "", //empty for now until we figure out why we need it
+					Status:             true,
+					State:              "Created",
+				}
+
+				// append new condition to conditions array
+				groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+				// update GroupPermission object with new conditions
+				instance.Status.Conditions = groupPermissionConditions
+
+				err = r.client.Status().Update(context.TODO(), instance)
+				if err != nil {
+					reqLogger.Error(err, "Failed to update condition.")
+					return reconcile.Result{}, err
```

Comment:
Need to make sure that if the status update is successful, the reconcile loop is still exited due to the creation fail.